### ISSUE TITLE
Unable to set mime type for a file when writing ?

### ISSUE THREAD

**Issue by faizanakram99** [thephpleague/flysystem#1424]
### Bug Report

|    Q        |   A
|------------ | ------
| BC Break    | yes/no
| Version     | 2.4.2

#### Summary

In v1 passing `['mimetype' => 'some mime']` would set mime type for uploaded files (whether `write` or `writeStream`) but this doesn't seem to work in v2. All files uploaded have mime type as `application/octet-stream`

Documentation only says `Filesystem::write()` accepts third parameter as `array $config`, can't find anywhere what are the possible config options, neither in documentation nor any hint from code.

I think the only thing that is needed is what are the possible values for `config array`, i think it might have changed in v2 but can't find it.

#### How to reproduce

```
$filesystem->write('some-file', 'something', ['mimetype' => 'text/plain']); // mime type is application/octet-stream instead of given mime type for aws s3 adapter
```

**Comment by frankdejonge** [comment]
Hi, which adapter are you using?

**Comment by faizanakram99** [comment]
> Hi, which adapter are you using?

AWS S3 V3 version 2.x

We don't use .txt or .pdf as filenames on server, on server the filenames are random integers or uuids and hence we used to set mime type by guessing it with `finfo` before uploading.
(think the build in mime type detector doesn't work well with such file names)

**Comment by frankdejonge** [comment]
the mime-type guesser also uses `finfo`. Nevertheless, I've just released 2.4.3 and 3.0.8 which allow you to set the mime-type like in v1.

**Comment by faizanakram99** [comment]
> the mime-type guesser also uses `finfo`. Nevertheless, I've just released 2.4.3 and 3.0.8 which allow you to set the mime-type like in v1.

Thank you very much
